scripts/reddit.js

```diff
@@ -0,0 +1,78 @@
+// Description:
+//   Grab top non-stickied post of a subreddit.
+//
+// Dependencies:
+//   reddit
+//
+// Configuration:
+//   HUBOT_REDDIT_USERNAME
+//   HUBOT_REDDIT_PASSWORD
+//   HUBOT_REDDIT_APP_ID
+//   HUBOT_REDDIT_APP_SECRET
+//
+// Commands:
+//   hubot reddit [subreddit] - Displays top non-stickied post of [subreddit], defaults to r/popular
+//
+// Notes:
+//   None.
+//
+// Author:
+//   https://github.com/davidscholberg
+
+const Reddit = require('reddit');
+
+function configureReddit() {
+  if (
+    typeof process.env.HUBOT_REDDIT_USERNAME === 'undefined' ||
+    typeof process.env.HUBOT_REDDIT_PASSWORD === 'undefined' ||
+    typeof process.env.HUBOT_REDDIT_APP_ID === 'undefined' ||
+    typeof process.env.HUBOT_REDDIT_APP_SECRET === 'undefined'
+  ) {
+    console.error(`error: missing one or more reddit configuration values`);
+    return null;
+  }
+  return new Reddit({
+    username: process.env.HUBOT_REDDIT_USERNAME,
+    password: process.env.HUBOT_REDDIT_PASSWORD,
+    appId: process.env.HUBOT_REDDIT_APP_ID,
+    appSecret: process.env.HUBOT_REDDIT_APP_SECRET,
+  });
+}
+
+module.exports = (robot) => {
+  const reddit = configureReddit();
+
+  robot.respond(/reddit( .+)?$/i, async (msg) => {
+    if (reddit === null) {
+      msg.send('oof, the reddit module is not configured properly :(');
+      return;
+    }
+    const subredditMatch = msg.match[1];
+    let subreddit = '';
+    if (typeof subredditMatch !== 'undefined' && subredditMatch.trim() !== '') {
+      subreddit = subredditMatch.trim();
+    } else {
+      subreddit = 'popular';
+    }
```

Comment:
Hmm, yeah I guess I never really liked ternaries lol. I always thought of the if-else form as more readable, but I can see the usefulness of consolidating it down to one line.

Also, I see your point about only doing positive checks, especially since the fact that the type is not `undefined` does not entail that it's a string, even though the code makes that assumption lol.

For now, I'll leave these as is, but at some point I'll want to go back and clean up small issues like this.